packages/loader/driver-utils/src/network.ts

```diff
@@ -43,6 +43,15 @@ export class GenericNetworkError extends LoggingError implements IDriverErrorBas
     }
 }
 
+export class SessionForbiddenError extends LoggingError implements IDriverErrorBase {
+    readonly errorType = DriverErrorType.sessionForbidden;
+    readonly canRetry = false;
```

Comment:
I do not know if it's good idea. But we should definitely consider converting canRetry from boolean to enum with these meanings:
- can retry
- fatal error, can't retry
- not available.
I.e. the last one is more a signal to the consumers of this error that while error will not go away, consumer should not treat as error, more as a state that does not support that thing.

That way we do not need to have special code for sessionForbidden.
And maybe some other paths could use same thing.
I.e. could we get rid of docService.policies?.storageOnly and instead use same mechanism to communicate you can't have delta connection?
Also question to @anthony-murphy (pushing this idea even further) - why DeltaManager needs to know about NoDeltaStream? Should it be moved to driver-utils package and returned by driver whenever it needs it?
It just does not feel like we need to plummet through these things when driver has all it needs to not leak these things to loader layer.

Reply:
FYI - I've opened https://github.com/microsoft/FluidFramework/issues/6050 to re-evaluate it as a separate task